### PR TITLE
Add a min docker version check

### LINKED ISSUE #1297: Cannot launch weave instances
I installed weave by doing a `sudo curl -L git.io/weave -o /usr/local/bin/weave`. Now I get this:
```
Abhishek@network:~$ weave launch
Unable to find image 'weaveworks/weaveexec:1.0.1' locally
Pulling repository weaveworks/weaveexec
8916b0188c16: Download complete
511136ea3c5a: Download complete
a5b60fe97da5: Download complete
16be31c9facc: Download complete
18e79490652f: Download complete
71b52c78f3a5: Download complete
a51ca901060c: Download complete
f2ccfd4b1f7f: Download complete
5baec2af0f8c: Download complete
d2a7854aa09a: Download complete
f5c600740af3: Download complete
3cf71612795c: Download complete
486bcb44ffa9: Download complete
Unable to find image 'weaveworks/weave:1.0.1' locally
2015/08/13 18:49:20 Error: client and server don't have same version (client : 1.15, server: 1.12)
Abhishek@network:~$ docker info
Containers: 0
Images: 13
Storage Driver: aufs
 Root Dir: /var/lib/docker/aufs
 Dirs: 13
Execution Driver: native-0.2
Kernel Version: 3.19.0-25-generic
WARNING: No swap limit support
Abhishek@network:~$ docker version
Client version: 1.0.1
Client API version: 1.12
Go version (client): go1.2.1
Git commit (client): 990021a
Server version: 1.0.1
Server API version: 1.12
Go version (server): go1.2.1
Git commit (server): 990021a
```
Also, I can run docker containers just fine. What am I doing wrong?

### PR DESCRIPTION
**Pull request by achanda**
Bailing out early is better UX
Properly fixes #1297

### CODE REVIEW

**Comment by rade** [comment]
This checks the Docker *client* version. It's the *server* version that matters. See https://github.com/weaveworks/weave/issues/576#issuecomment-94940727 for the gory details (and why we haven't done this).

Though perhaps mandating client >=1.3.1 is the least worst option here.

File: weave [hunk 1]
```diff
@@ -102,13 +103,15 @@ check_docker_version() {
         exit 1
     fi
 
-    # guard against https://github.com/docker/docker/issues/8632
-    if [ "$DOCKER_VERSION" = "1.3.0" ] ; then
-        echo "You are running Docker version $DOCKER_VERSION, which contains a bug that prevents weave from working properly. Please upgrade." >&2
+    FLAG=$(awk 'BEGIN { print ("$DOCKER_VERSION" < "$MIN_DOCKER_VERSION") ? "1" : "0" }')
```
Comment (rade) [inline weave]: This does not produce the correct result:

````
$ awk 'BEGIN { print ("1.10.0" < "1.3.1") ? "1" : "0" }'
1
````

Comment (achanda) [inline weave]: This indeed fails for version numbers of the form x.y.z. I will try to come up with something that works better.

**Comment by rade** [comment]
superseded by  #1326.

**Pull request closed without merging**